pkg/distributors/contracts/MultiRewards.sol

```diff
@@ -478,19 +477,28 @@ contract MultiRewards is IMultiRewards, IDistributor, ReentrancyGuard, MultiRewa
         IERC20 token
     ) internal {
         uint256 totalUnpaidRewards;
-        for (uint256 r; r < _rewarders[pool][token].length(); r++) {
-            address rewarder = _rewarders[pool][token].unchecked_at(r);
 
-            rewardData[pool][rewarder][token].rewardPerTokenStored = rewardPerToken(pool, rewarder, token);
-            rewardData[pool][rewarder][token].lastUpdateTime = lastTimeRewardApplicable(pool, rewarder, token);
+        // Save the storage pointer to compute the slot only once.
+        EnumerableSet.AddressSet storage rewarders = _rewarders[pool][token];
+        uint256 rewardersLength = rewarders.length();
+
+        for (uint256 r; r < rewardersLength; r++) {
+            address rewarder = rewarders.unchecked_at(r);
+            Reward storage data = rewardData[pool][rewarder][token];
+
+            // Cache storage variables to avoid repeated access.
+            uint256 perToken = rewardPerToken(pool, rewarder, token);
```

Comment:
We're only using `pool`, `rewarder` and `token` here to recompute `rewardData[pool][rewarder][token]`. We could make an internal `_rewardPerToken(Reward storage data)` to avoid this.

Similarly in `lastTimeRewardApplicable`.

Reply:
We're also just using a one or two fields from `data` and not writing anything back to storage in those functions so could just pass those fields in directly.

Reply:
Other than this small possible improvement I'm happy with it.

Reply:
Oh this is very good, I didn't notice those functions accessed the same data structure. Will include this.